climada/hazard/relative_cropyield.py

```diff
@@ -0,0 +1,710 @@
+Software Foundation, version 3.
+
+CLIMADA is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with CLIMADA. If not, see <https://www.gnu.org/licenses/>.
+
+---
+
+Define AgriculturalDrought (AD) class.
+WORK IN PROGRESS
+"""
+
+__all__ = ['RelativeCropyield']
+
+import logging
+import os
+from os import listdir
+from os.path import isfile, join
+import numpy as np
+from matplotlib import pyplot as plt
+import cartopy
+import shapely.geometry
+from scipy import sparse
+import scipy.stats
+import h5py
+
+
+from climada.hazard.base import Hazard
+from climada.util import dates_times as dt
+from climada.util import coordinates
+from climada.util.constants import DATA_DIR
+
+
+LOGGER = logging.getLogger(__name__)
+
+HAZ_TYPE = 'RC'
+"""Hazard type acronym for Relative Cropyield"""
+
+INT_DEF = 'Yearly Yield'
+
+BBOX = np.array([-180, -85, 180, 85])  # [Lon min, lat min, lon max, lat max]
+""""Default geographical bounding box of the global agricultural land extent"""
+
+# ! deposit the input files in: climada_python/data/ISIMIP_crop/Input/Hazard
+INPUT_DIR = os.path.join(DATA_DIR, 'ISIMIP_crop', 'Input', 'Hazard')
+"""default paths for input and output data:"""
+OUTPUT_DIR = os.path.join(DATA_DIR, 'ISIMIP_crop', 'Output')
+
+
+#ISIMIP input data specific global variables
+YEARCHUNKS = dict()
+"""start and end years per senario as in ISIMIP-filenames"""
+YEARCHUNKS['ISIMIP2a'] = dict()
+YEARCHUNKS['ISIMIP2a'] = {'yearrange': np.array([1980, 1999]), 'startyear': 1980,
+                          'endyear': 1999, 'yearrange_mean': np.array([1980, 1999])}
+YEARCHUNKS['historical'] = dict()
+YEARCHUNKS['historical'] = {'yearrange': np.array([1976, 2005]), 'startyear': 1861,
+                            'endyear': 2005, 'yearrange_mean': np.array([1976, 2005])}
+YEARCHUNKS['rcp60'] = dict()
+YEARCHUNKS['rcp60'] = {'yearrange': np.array([2006, 2099]), 'startyear': 2006,
+                       'endyear': 2099}
+
+FN_STR_VAR = 'global_annual'
+"""filename of ISIMIP output constant part"""
+
+
+class RelativeCropyield(Hazard):
+    """Agricultural climate risk: Relative Cropyield (relative to historical mean);
+    Each year corresponds to one hazard event;
+    Based on modelled crop yield, from ISIMIP (www.isimip.org, required input data).
+
+    Attributes:
+        crop_type (str): crop type (e.g. whe for wheat)
+        intensity_def (str): intensity defined as:
+            'Yearly Yield' [t/(ha*y)], 'Relative Yield', or 'Percentile'
+    """
+
+    def __init__(self, pool=None):
+        """Empty constructor."""
+        Hazard.__init__(self, HAZ_TYPE)
+        if pool:
+            self.pool = pool
+            LOGGER.info('Using %s CPUs.', self.pool.ncpus)
```

Comment:
Is pool used anywhere or is it for future use?

Reply:
It is not used currently.
Therefore I suggest that we replace

```
        if pool:
            self.pool = pool
            LOGGER.info('Using %s CPUs.', self.pool.ncpus)
        else:
            self.pool = None
```

With:
`self.pool = None`

Reply:
Actually, it is not a big deal and maybe should be kept for the future as it is now... It was just to know.